src/main/java/org/intellij/sonar/configuration/SonarQualifier.java

```diff
@@ -1,18 +1,29 @@
 package org.intellij.sonar.configuration;
 
-import org.sonar.wsclient.services.Resource;
+import static java.util.Arrays.stream;
+import static java.util.function.Function.identity;
+import static java.util.stream.Collectors.toMap;
+
+import java.util.Map;
 
 public enum SonarQualifier {
-  PROJECT,MODULE;
+    PROJECT("TRK"), MODULE("BRC");
+
+    private static Map<String, SonarQualifier> instanceByQualifier;
+    private String qualifier;
+
+    SonarQualifier(String qualifier) {
+        this.qualifier = qualifier;
+    }
+
+    public String getQualifier() {
+        return qualifier;
+    }
 
-  public static SonarQualifier valueFrom(String qualifier) {
-    if (Resource.QUALIFIER_PROJECT.equals(qualifier)) {
-      return PROJECT;
-    } else
-      if (Resource.QUALIFIER_MODULE.equals(qualifier)) {
-        return MODULE;
-      } else {
-        throw new IllegalArgumentException(qualifier);
-      }
-  }
+    public static SonarQualifier valueFrom(String qualifier) {
+        if (instanceByQualifier == null) {
+            instanceByQualifier = stream(values()).collect(toMap(SonarQualifier::getQualifier, identity()));
+        }
+        return instanceByQualifier.get(qualifier);
```

Comment:
if we cannot find the qualifier, it will return null, causing NullPointerException.
Throwing another exception like IlligalArgumentException makes sense to me

Reply:
as you wish